compiler/src/dotty/tools/dotc/transform/PostTyper.scala

```diff
@@ -255,15 +257,19 @@ class PostTyper extends MacroTransform with IdentityDenotTransformer { thisPhase
 
     override def transform(tree: Tree)(using Context): Tree =
       try tree match {
-        case tree: Ident if !tree.isType =>
-          if tree.symbol.is(Inline) && !Inliner.inInlineMethod then
-            ctx.compilationUnit.needsInlining = true
-          checkNoConstructorProxy(tree)
-          tree.tpe match {
-            case tpe: ThisType => This(tpe.cls).withSpan(tree.span)
-            case _ => tree
-          }
+        case tree: Ident =>
+          Experimental.checkExperimental(tree)
```

Comment:
We briefly discussed doing these checks in RefChecks recently and I think that indeed makes sense, in fact it might be possible to add them to the existing `checkDeprecated` and `checkDeprecatedOvers` methods.

Reply:
I tried, it is much harder to do it in `checkDeprecated` and `checkDeprecatedOvers` do no cover all the trees and a many extra special cases would be needed.

Reply:
> These do no cover all the trees and a many extra special cases would be needed.

Do you have an example? Wouldn't that mean that the deprecation checking is incomplete?

Reply:
One case I remember was that the check did not cover annotations. I will try it again.